roxctl/central/generate/k8s.go

```diff
@@ -176,6 +176,7 @@ func openshift(cliEnvironment environment.Environment) *cobra.Command {
 	flagWrap.Var(&fileFormatWrapper{DeploymentFormat: &k8sConfig.DeploymentFormat}, "output-format", fmt.Sprintf("the deployment tool to use (%s)", strings.Join(validFormats, ", ")), "central")
 
 	flagWrap.IntVar(&openshiftVersion, "openshift-version", 0, "the OpenShift major version (3 or 4) to deploy on")
+	flagWrap.BoolVar(&k8sConfig.Monitoring.OpenShiftMonitoring, "openshift-monitoring", false, "whether to enable integration with OpenShift monitoring (requires OpenShift 4)", "central")
```

Comment:
> requires OpenShift 4

How about adding a validation step for this? So if user enables monitoring but do not set OpenShift version to 4 let's show warning or fail.

Reply:
The reason is that then either we cannot set `openshift-monitoring=true` by default, or it will always fail with the default settings. The fact that `openshift-version=3` is the default makes the situation quite awkward.

We already handle `openshift-version=3` via the override in the Helm chart, which disables openshift monitoring. So I'm not sure we need another validation in roxctl.

I think the question is if we want `roxctl central generate openshift --openshift-version=4` to have monitoring enabled by default, and if we are fine with Helm warning and implicit override in case of `roxctl central generate openshift --openshift-version=3 --openshift-monitoring=true`, or if we want a hard fail. Basically if we want a fail with validation, we need to put `openshift-monitoring=false` as default.